addons/l10n_nl_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,150 @@
+        values = super()._get_ubl_values(invoice)
+        if self.code != 'nlcius_1':
+            return values
+
+        values['customization_id'] = 'urn:cen.eu:en16931:2017#compliant#urn:fdc:nen.nl:nlcius:v1.0'
+        values['payment_means_code'] = 30
+        values['partner_template'] = 'l10n_nl_edi.export_en_16931_invoice_partner'
+        return values
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_nlcius(self, filename, tree):
+        if self.code != 'nlcius_1':
+            return False
+        customization_id = self._find_value("//*[local-name()='CustomizationID']", tree)
+        return 'nlcius' in customization_id
+
+    def _decode_en_16931(self, tree, invoice):
+        res = super()._decode_en_16931(tree, invoice)
+        if self.code != 'nlcius_1':
+            return res
+
+        if not res.partner_id:
+            namespaces = self._get_ubl_namespaces(tree)
+            endpoint = tree.xpath('//cac:AccountingSupplierParty/cac:Party//cbc:EndpointID', namespaces=namespaces)
+            if endpoint:
+                endpoint = endpoint[0]
+                scheme = endpoint.attrib['schemeID']
+                domains = []
+                if scheme == '0106' and endpoint.text:
+                    domains.append([('l10n_nl_kvk', '=', endpoint.text)])
+                elif scheme == '0190' and endpoint.text:
+                    domains.append([('l10n_nl_oin', '=', endpoint.text)])
+                if domains:
+                    partner = self.env['res.partner'].search(expression.OR(domains), limit=1)
+                    if partner:
+                        res.partner_id = partner
+        return res
+
+    ####################################################
+    # Export
+    ####################################################
+
+    def _check_move_configuration(self, invoice):
+        res = super()._check_move_configuration(invoice)
+        if self.code != 'nlcius_1':
+            return res
+
+        errors = self._check_en_16931_invoice_configuration(invoice)
+
+        supplier = invoice.company_id.partner_id.commercial_partner_id
+        if not supplier.street or not supplier.zip or not supplier.city:
+            errors.append(_("The supplier's address must include street, zip and city."))
+        if supplier.country_code == 'NL' and not supplier.l10n_nl_kvk and not supplier.l10n_nl_oin:
+            errors.append(_("The supplier must have a KvK-nummer or OIN."))
+
+        customer = invoice.commercial_partner_id
+        if customer.country_code == 'NL' and (not customer.street or not customer.zip or not customer.city):
+            errors.append(_("Customer's address must include street, zip and city."))
+        if customer.country_code == 'NL' and not customer.l10n_nl_kvk and not customer.l10n_nl_oin:
+            errors.append(_("The customer must have a KvK-nummer or OIN."))
```

Comment:
you should consider to inject the `display_name` inside your errors `errors.append(_("The customer %s must have a KvK-nummer or OIN.", customer.display_name))`